webapp/channels/src/components/setting_item_max.test.tsx

```diff
@@ -24,134 +23,86 @@ describe('components/SettingItemMax', () => {
-        const wrapper = shallow(<SettingItemMax {...props}/>);
-        const instance = wrapper.instance() as SettingItemMax;
-        const event = {
-            preventDefault: jest.fn(),
-        } as unknown as React.MouseEvent<HTMLButtonElement>;
+    test('should have called submit on handleSubmit onKeyDown ENTER', () => {
+        const props = {
+            ...baseProps,
+            inputs: [
+                <select
+                    key={0}
+                    data-testid='select'
+                />,
+                <input
+                    key={1}
+                    type='radio'
+                />,
+            ],
+        };
 
-        instance.handleSubmit(event);
+        const {getByRole, getByTestId} = renderWithContext(<SettingItemMax {...props}/>);
 
-        expect(submit).toHaveBeenCalled();
-        expect(submit).toHaveBeenCalledWith();
-    });
+        getByTestId('select').focus();
 
-    it('should have called submit on handleSubmit onKeyDown ENTER', () => {
-        const submit = jest.fn();
-        const props = {...baseProps, submit};
-        const wrapper = shallow(<SettingItemMax {...props}/>);
-        const instance = wrapper.instance() as SettingItemMax;
-
-        instance.onKeyDown({
-            preventDefault: jest.fn(),
-            key: Constants.KeyCodes.ENTER[0],
-            target: {
-                tagName: 'SELECT',
-                classList: {contains: jest.fn()},
-                parentElement: {className: 'react-select__input'},
-            },
-        } as unknown as KeyboardEvent);
-
-        expect(submit).toHaveBeenCalledTimes(0);
-
-        instance.settingList = {
-            current: {
-                contains: jest.fn(() => true),
-            } as unknown as HTMLDivElement,
-        };
-        instance.onKeyDown({
-            preventDefault: jest.fn(),
-            key: Constants.KeyCodes.ENTER[0],
-            target: {
-                tagName: '',
-                classList: {contains: jest.fn()},
-                parentElement: {className: ''},
-            },
-        } as unknown as KeyboardEvent);
-
-        expect(submit).toHaveBeenCalledTimes(1);
-        expect(submit).toHaveBeenCalledWith('setting');
-    });
+        /**
+         * RTL recommends to use this approach to test keydown events.
+         * https://testing-library.com/docs/guide-events/#keydown
+         */
+        fireEvent.keyDown(document.activeElement!, {key: 'Enter', code: 'Enter'});
+        expect(baseProps.submit).toHaveBeenCalledTimes(0);
 
-    test('should match snapshot, with new saveTextButton', () => {
-        const props = {...baseProps, saveButtonText: 'CustomText'};
-        const wrapper = shallow(<SettingItemMax {...props}/>);
+        getByRole('radio').focus();
 
-        expect(wrapper).toMatchSnapshot();
+        fireEvent.keyDown(document.activeElement!, {key: 'Enter', code: 'Enter'});
+        expect(baseProps.submit).toHaveBeenCalledTimes(1);
     });
 });
```

Comment:
I swapped the last test with the newly added test because, for some reason, focusing and blurring the input resulted in the `SaveButton` of the previously last test (`'should match snapshot, with new saveTextButton'`) to be focused i.e. it gets the `a11y` focus class.

Placing this newly added test anywhere but above the previously last test worked fine, and all tests pass.

The focus class is what changes the snapshot, otherwise, everything else is okay.